url36deadcd05_SCass1986_spawncamping_cyril_tgz-pJ8-org_stephen_hashmap_MainJ8/src/org/stephen/hashmap/Logger.java

```diff
@@ -0,0 +1,36 @@
+package org.stephen.hashmap;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class Logger {
+    private static final String LOG_STRING = "[%s] %-10s: ";
+    private static final String INFO       = "INFO";
+    private static final String DEBUG      = "DEBUG";
+    private static final String VERBOSE    = "VERBOSE";
+
+    private static final Calendar calendar = Calendar.getInstance ();
+
+    public static void info (final String message, final Object... args) {
+        if (Main.LOGGING_INFO_ON) {
+            System.out.println (String.format ("%s%s", String.format (LOG_STRING, getTimeStamp (), INFO), String.format (message, args)));
+        }
+    }
+
+    public static void verbose (final String message, final Object... args) {
+        if (Main.LOGGING_VERBOSE_ON) {
+            System.out.println (String.format ("%s%s", String.format (LOG_STRING, getTimeStamp (), INFO), String.format (message, args)));
+        }
+    }
+
+    public static void debug (final String message, final Object... args) {
+        if (Main.LOGGING_DEBUG_ON) {
+            System.out.println (String.format ("%s%s", String.format (LOG_STRING, getTimeStamp (), DEBUG), String.format (message, args)));
+        }
+    }
+
+    private static String getTimeStamp () {
+        return new SimpleDateFormat ("dd/mm/yyyy hh:MM:ss").format (calendar.getTime ());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.